cl/stmt.go

```diff
@@ -333,6 +341,44 @@ func compileForPhraseStmt(ctx *blockCtx, v *ast.ForPhraseStmt) {
 	cb.End()
 }
 
+func toForStmt(forPos token.Pos, value ast.Expr, body *ast.BlockStmt, re *ast.RangeExpr) *ast.ForStmt {
+	if value == nil || value.(*ast.Ident).Name == "_" {
+		value = &ast.Ident{NamePos: forPos, Name: "_gop_k"}
+	}
+	if re.First == nil {
+		re.First = &ast.BasicLit{ValuePos: forPos, Kind: token.INT, Value: "0"}
+	}
+	if re.Expr3 == nil {
+		re.Expr3 = &ast.BasicLit{ValuePos: forPos, Kind: token.INT, Value: "1"}
+	}
+	return &ast.ForStmt{
+		For: forPos,
+		Init: &ast.AssignStmt{
+			Lhs:    []ast.Expr{value},
+			TokPos: re.To,
+			Tok:    token.DEFINE,
+			Rhs:    []ast.Expr{re.First},
+		},
+		Cond: &ast.BinaryExpr{
+			X:     value,
+			OpPos: re.To,
+			Op:    token.LSS,
+			Y:     re.Last,
```

Comment:
maybe `re.Last` is a `function call`. it's not reasonable to call `re.Last` multiple times.